src/conversion/viewconversiondispatcher.js

```diff
@@ -180,6 +179,30 @@ export default class ViewConversionDispatcher {
 			this.fire( 'documentFragment', data, consumable, this.conversionApi );
 		}
 
+		// Handle incorrect `data.output`.
+		if ( data.output === undefined ) {
+			/**
+			 * Item has not been converted (output is undefined).
+			 *
+			 * @error view-conversion-dispatcher-item-not-converted
+			 */
+			log.warn( 'view-conversion-dispatcher-item-not-converted: Item has not been converted (output is undefined).', input );
+
+			data.output = null;
+		} else if ( !( data.output instanceof ModelNode || data.output instanceof ModelDocumentFragment ) ) {
```

Comment:
I wanted to comment that it'd be better if we used duck typing with `is()` as it's safer. But I can see that we use a lot of instanceofs around our code still so we can change this at once in the future.